scripts/validate_data_integrity.py

```
@@ -2,22 +2,158 @@
+            SELECT COUNT(*) FROM (
+                SELECT event_id, COUNT(DISTINCT tenant_id) as tenant_count
+                FROM fsma.cte_events
+                GROUP BY event_id
+                HAVING COUNT(DISTINCT tenant_id) > 1
+            ) violations
+        """))
+        count = result.scalar()
+        if count and count > 0:
+            return False, f"CRITICAL: {count} events found in multiple tenants"
+        return True, "Tenant isolation verified — no cross-tenant leakage"
+    except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)